src/hooks/FetchData.jsx

```diff
@@ -0,0 +1,38 @@
+import { useDispatch } from "react-redux";
+import { SET_LOADING } from "../redux/fake-api-data/fakeApiDataTypes";
+import FormatData from "../utils/FormatApiData";
+
+const FetchData = () => {
+  const dispatch = useDispatch();
+
+  const fetchDataFn = async () => {
+    dispatch({
+      type: SET_LOADING,
+      loading: true,
+    });
+
+    try {
+      const res = await fetch("https://fakestoreapi.com/products");
+      const data = await res.json();
+
+      const formattedData = FormatData(data);
+
+      dispatch({
+        type: SET_LOADING,
+        loading: false,
+      });
+
+      return formattedData;
+    } catch (error) {
+      console.log("Error in fetchData!", error);
+      dispatch({
+        type: SET_LOADING,
+        loading: false,
```

Comment:
@9xVibee instead of having loader false here and in try block you can move this code to finally